notifications/metrics.go

```diff
@@ -56,24 +58,36 @@ type endpointMetricsHTTPStatusListener struct {
 
 var _ httpStatusListener = &endpointMetricsHTTPStatusListener{}
 
-func (emsl *endpointMetricsHTTPStatusListener) success(status int, events ...Event) {
+func (emsl *endpointMetricsHTTPStatusListener) success(status int, event events.Event) {
 	emsl.safeMetrics.Lock()
 	defer emsl.safeMetrics.Unlock()
-	emsl.Statuses[fmt.Sprintf("%d %s", status, http.StatusText(status))] += len(events)
-	emsl.Successes += len(events)
+	evt, ok := event.(Event)
+	if !ok || (ok && evt.ID == "") {
```

Comment:
Does this happen? If we are seeing this, there might be a bug.

Reply:
I believe this shouldn't happen.  From memory I think I added this after I accidentally sent the wrong thing through here partway through converting things.  I'll remove it.